cli/src/commands/branch.rs

```diff
@@ -316,6 +303,23 @@ fn cmd_branch_rename(
         ),
     )?;
 
+    let view = workspace_command.repo().view();
+    if view
+        .remote_branches_matching(
+            &StringPattern::exact(old_branch),
+            &StringPattern::everything(),
+        )
+        .any(|(_, remote_ref)| remote_ref.is_tracking())
+    {
+        writeln!(
+            ui.warning(),
+            "Warning: Branch {old_branch} is tracking remote branches which were not \
+             renamed.\nUse `jj git push --tracked` to have a remote delete the branch with the \
+             old name and create a new branch with the new name. This will also push any other \
```

Comment:
I don't think `--tracked` creates a new remote branch.

Reply:
Good point. :(

Reply:
I removed the commit from this PR for now, might submit a version of it later. For reference, it's available at https://github.com/ilyagr/jj/tree/push-tracked-misc for now.